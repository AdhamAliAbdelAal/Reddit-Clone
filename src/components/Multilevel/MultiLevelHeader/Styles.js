import { Box, Button } from '@mui/material';
import { styled } from '@mui/material/styles';
import CloseIcon from '@mui/icons-material/Close';
// MultiLevel Container
export const MultiLevelHeaderBox = styled(Box)(() => ({
  flexBasis: '100%',
  backgroundColor: 'black',

  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
}));

export const CloseButton = styled(() => (
  <Button variant="text" startIcon={<CloseIcon fontSize="small" />} color="third">Close</Button>
))(() => ({
//   '&.MuiButton-text': {
//     textTransform: 'capitalize',
//     backgroundColor: 'red',
//   },
}));

export const MultiLevelHeaderVotes = styled(Box)(() => ({
  display: 'flex',
  flexDirection: 'row',
  justifyContent: 'space-between',
}));
