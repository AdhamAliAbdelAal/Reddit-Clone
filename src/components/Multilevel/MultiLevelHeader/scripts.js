export const addPostToHistory = (post) => {
  // isHidden
  const savedPost = (({
    _id, author, owner, ownerType,
    title, kind,
    text, url, votes, commentCount, views, createdAt, suggestedSort, nsfw, spoiler,
    sendReplies, locked, flairId, flairText, postVoteStatus, isSaved, isHidden, modState, sharedFrom, shareCount, images, video, sortOnHot,
  }) => ({
    _id,
    author,
    owner,
    ownerType,
    title,
    kind,
    text,
    url,
    votes,
    commentCount,
    views,
    createdAt,
    suggestedSort,
    nsfw,
    spoiler,
    sendReplies,
    locked,
    flairId,
    flairText,
    postVoteStatus,
    isSaved,
    isHidden,
    modState,
    sharedFrom,
    shareCount,
    images,
    video,
    sortOnHot,
  }))(post);
  savedPost.isHidden = false;
  console.log('Saved Post');
  console.log(savedPost);

  // Push This post to Hiistory in the local storage
  const savedPostArray = [savedPost];
  let history;
  if (localStorage.getItem('RedditHistory')) {
    const historyArray = JSON.parse(localStorage.getItem('RedditHistory'));
    // Remove From array
    const objWithIdIndex = historyArray.findIndex((obj) => obj._id === savedPost._id);
    if (objWithIdIndex > -1) {
      historyArray.splice(objWithIdIndex, 1);
    }

    history = [...savedPostArray, ...historyArray];
  } else {
    history = [...savedPostArray];
  }
  console.log(history);
  localStorage.setItem('RedditHistory', JSON.stringify(history));
};
