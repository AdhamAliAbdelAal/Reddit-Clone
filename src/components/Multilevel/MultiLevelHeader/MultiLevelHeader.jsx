import { useEffect } from 'react';
import { useNavigate } from 'react-router-dom';

// MUI components
import {
  Typography, Button, IconButton,
} from '@mui/material';

// Icons
import CloseIcon from '@mui/icons-material/Close';
import ArticleIcon from '@mui/icons-material/Article';

// Components
import VoteIcon from '../../VoteIcons/VoteIcon/VoteIcon';
// import VotedIcon from '../../VoteIcons/VotedIcon/VotedIcon';

// Contexts
import { usePostContext } from '../../../contexts/PostContext';

// Styles
import {
  HeaderVerticalDivider, MultiLevelHeaderBox, MultiLevelHeaderVotes, TopPostVotesConatiner,
} from './styles';

// Theme
import theme from '../../../styles/theme';
import { addPostToHistory } from './scripts';

function MultiLevelHeader() {
  // Context
  const { post } = usePostContext();

  // useNavigate
  const navigate = useNavigate();
  // useEffect
  useEffect(() => {
    console.log('MultiLevelHeader.jsx', post);
    // Add This post to the History
    if (post !== null) {
      addPostToHistory(post);
    }
  }, []);

  const close = () => {
    // Backward
    navigate(-1);
  };

  const upVote = () => {
    console.log('UpVote');
  };

  const downVote = () => {
    console.log('DownVote');
  };
  return (
    post
    && (
    <MultiLevelHeaderBox>
      <MultiLevelHeaderVotes>
        <TopPostVotesConatiner>
          {/* <Divider orientation="vertical" flexItem /> */}
          <HeaderVerticalDivider orientation="vertical" flexItem />
          <IconButton onClick={upVote}>
            <VoteIcon
              color={theme.palette.secondary?.main}
              direction="up"
            />
          </IconButton>
          <Typography variant="p">{post?.votes}</Typography>
          <IconButton onClick={downVote}>
            <VoteIcon
              color={theme.palette.secondary?.main}
              direction="down"
            />
          </IconButton>
          <HeaderVerticalDivider orientation="vertical" flexItem />
          {/* <VotedIcon
          color={theme.palette.secondary?.main}
          direction="down"
        /> */}
        </TopPostVotesConatiner>
        <ArticleIcon />
        <Typography variant="p" fontSize="14px">{post?.title}</Typography>
      </MultiLevelHeaderVotes>
      <Button variant="text" startIcon={<CloseIcon fontSize="small" />} color="third" onClick={close}>Close</Button>

    </MultiLevelHeaderBox>
    )
  );
}

export default MultiLevelHeader;
