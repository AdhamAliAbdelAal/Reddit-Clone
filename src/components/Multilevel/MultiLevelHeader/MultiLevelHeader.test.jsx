import renderer from 'react-test-renderer';
import '@testing-library/jest-dom';

import PostContextProvider from '../../../contexts/PostContext';
import MultiLevelHeader from './MultiLevelHeader';

// Use Navigate
const mockedUsedNavigate = jest.fn();
jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useNavigate: () => mockedUsedNavigate,
}));

// test snapshot
test('MultiLevelHeader-test-snapshot', async () => {
  // Render Component Of Post Provider
  const tree = renderer.create(
    <PostContextProvider postID="123556">
      <MultiLevelHeader />
    </PostContextProvider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});
