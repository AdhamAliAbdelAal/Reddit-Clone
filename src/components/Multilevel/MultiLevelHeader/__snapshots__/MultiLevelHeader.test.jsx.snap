// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MultiLevelHeader-test-snapshot 1`] = `
Object {
  "instance": null,
  "nodeType": "component",
  "props": Object {
    "children": <MultiLevelHeader />,
    "postID": "123556",
  },
  "rendered": Object {
    "instance": null,
    "nodeType": "component",
    "props": Object {},
    "rendered": null,
    "type": [Function],
  },
  "type": [Function],
}
`;
